### PR TITLE
Add basic i18n support for problem text

### PR DESCRIPTION
**Pull request by watilde**
Hi.
I added basic i18n support!

Here is updates:
- Switched to workshopper from adventure for i18n support
- Separate problem text from JavaScript file of problem: [patch](https://gist.github.com/watilde/f201fb146b394bf53003)
- Add i18n support

Preview:
image

Thanks!

### CODE REVIEW

**Comment by martinheidegger** [comment]
@watilde I would like to improve on this PR. Would it be okay to mark it as "work-in-progress"?

File: package.json [hunk 1]
```diff
@@ -20,20 +20,20 @@
...
   ],
   "dependencies": {
-    "adventure": "^2.10.0",
     "concat-stream": "^1.4.7",
     "mkdirp": "^0.5.0",
     "rimraf": "^2.2.8",
     "semver": "^4.2.0",
-    "which": "^1.0.8"
+    "which": "^1.0.8",
+    "workshopper-adventure": "^3.4.1"
```
Comment (martinheidegger) [inline package.json]: The latest workshopper is 5.0.3. It would be good to update it!

**Comment by watilde** [comment]
@martinheidegger Yes, it's ok! I've added WIP tag into the title.

File: problems/dev-environment/index.js [hunk 2]
```diff
@@ -1,31 +1,13 @@
...
-//cd test
-//npm init
-//# fill in all the prompts
-//
-//# Extra Credit:
-//git init
-//*/}.toString().split('\n').slice(1,-1).join('\n')
+exports.init = function (workshopper) {
+  var postfix = workshopper.lang === 'en' ? '' : '_' + workshopper.lang
+  this.problem = function () {
```
Comment (martinheidegger) [inline problems/dev-environment/index.js]: Note: it is possible to specify a problem like: 

```
problem = {
  file: path.join(__dirname, 'problem.{lang}.txt')
}
```

but of course this means that you would need to rename the existing .txt files

File: problems/dist-tag-removal/problem.txt [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+every package installs it's "latest" tag by default, so that tag has
+some special semantics.
+
+You CAN point "latest" to a different version, or delete other tags.
+
+Let's delete all the tags that we can, and also point "latest" at
+something other than the most recent release.
+
+Run `npm help dist-tag` to learn more about the command.
+*/}.toString().split('\n').slice(1,-1).join('\n')
```
Comment (martinheidegger) [inline problems/dist-tag-removal/problem.txt]: This line seems to be added by mistake

**Review by martinheidegger — CHANGES_REQUESTED** [review]
For the biggest part it looks okay to me. There are some things that could look smoother but basically it seems to work. A rebase to master would be good before I can do a technical test.

File: problems/dev-environment/index.js (hunk 2 above)
Comment (watilde) [reply problems/dev-environment/index.js]: That's good idea

**Comment by watilde** [comment]
@martinheidegger Hey Martin, I've rebased and got rid of translated problem texts to keep the diff more less

**Comment by watilde** [comment]
Merged https://github.com/watilde/how-to-npm/pull/1

**Comment by martinheidegger** [comment]
I added another PR to your branch with the missing exercises. Please also run npm test and fix standard formatting ^_^

**Comment by martinheidegger** [comment]
For extra points:
1. It might be nice to move to markdown instead of text files?
2. In `/i18n/en.json` i put all the exercises in the root node but it would be better if they were one level deeper in `exercises`
3. Maybe it would be a good idea to add this to the workshopper definition? _(Would need textual updates)_

``` javascript
    bg: 'white',
    fg: 'red'
  },
  header: require('workshopper-adventure/default/header'),
  footer: require('workshopper-adventure/default/footer'),
  commands: [{
```

**Comment by martinheidegger** [comment]
Do you want to do more changes?

**Comment by watilde** [comment]
@martinheidegger I've fixed the lint errors, and title names. Then let's do that!

**Comment by watilde** [comment]
`start-a-project/index.js` was old, and I've updated it. Verified ✅

File: menu.json [hunk 4]
```diff
@@ -1,18 +1,19 @@
...
-  "Publish",
-  "Version",
-  "Publish Again",
-  "Dist Tag",
-  "Dist Tag Removal",
-  "Outdated",
-  "Update",
-  "Rm",
-  "Finale"
+  "00 Install npm",
```
Comment (martinheidegger) [inline menu.json:2]: By changing this you should also change the i18n/en.json in which there are pointers to these names

Comment (watilde) [reply menu.json:2]: That's right. I've updated the menu.json as well

**Comment by watilde** [comment]
@martinheidegger Ok i've tried them all again. Added one fix https://github.com/workshopper/how-to-npm/pull/20/commits/bb111a69c2b6e3bcc183c822b22d8a03690cf1d7, and it worked fine eventually.

**Comment by martinheidegger** [comment]
@workshopper/core LGTM: ship it?!

**Review by martinheidegger — APPROVED** [review]

**Comment by watilde** [comment]
Heyy thanks JP and Martin!

**Comment by a0viedo** [comment]
sorry for getting somewhat late, but great work @watilde @martinheidegger @jpwesselink  :tada:

**Pull request merged**